tests/qinfo/test_utils_qinfo.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for quantum info utils functions."""
+
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+pytestmark = pytest.mark.all_interfaces
+
+tf = pytest.importorskip("tensorflow", minversion="2.1")
+torch = pytest.importorskip("torch")
+jax = pytest.importorskip("jax")
+
+
+angle_values = [0, np.pi / 4, np.pi / 2, 3 * np.pi / 4, np.pi]
+devices = [
+    "default.qubit",
+    "default.mixed",
+]
+interfaces = [
+    "autograd",
+    "torch",
+    "tf",
+    "jax",
+]
+wires_list = [[0], [1], [0, 1]]
+check_state = [False, True]
+
+
+class TestDensityMatrixQNode:
```

Comment:
It would be good to test (for this and all future additions) that the dtype of the density matrix matches the user-provided dtype of the device.